presto-main/src/main/java/io/prestosql/sql/planner/LogicalPlanner.java

```diff
@@ -418,8 +419,17 @@ private RelationPlan createTableWriterPlan(
 
             List<Symbol> outputLayout = new ArrayList<>(symbols);
 
+            PartitioningHandle partitioningHandle;
+            if (writeTableLayout.get().getLayout().getPartitioning().isPresent()) {
+                partitioningHandle = writeTableLayout.get().getPartitioning();
+            }
+            else {
+                // empty connector partitioning handle means evenly partitioning on partitioning columns
+                partitioningHandle = FIXED_HASH_DISTRIBUTION;
+            }
```

Comment:
I find the fact we're looking at `getLayout().getPartitioning()` to decide whether to call `getPartitioning()` very confusing, and probably a sign of an API design issue.

I'd be cleaner to make `getPartitioning()` for `NewTableLayout` return optional and just do:

```java
PartitioningHandle partitioningHandle = writeTableLayout.get()
    .getPartitioning()
    .orElse(FIXED_HASH_DISTRIBUTION)
```